calculate_trigger.py

```diff
@@ -186,16 +186,24 @@ def calculate_trigger_addresses(fault_list, goldenrun_tb_exec, goldenrun_tb_info
             if found is True:
                 continue
 
-            tbs = search_for_fault_location(
-                lists,
-                fault.trigger.address,
-                fault.address,
-                fault.trigger.hitcounter,
-                fault.lifespan,
-                goldenrun_tb_exec,
-                goldenrun_tb_info,
-            )
-            d = {}
+            if fault.lifespan != 0 and fault.trigger.address + fault.lifespan < 0:
```

Comment:
We should avoid caching invalid faults. Please move this check to the top, to line [172](https://github.com/Fraunhofer-AISEC/archie/blob/d8d880a5a5c51235890b001f72f12fe2c6ae9ce5/calculate_trigger.py#L172)

Reply:
This can lead to many warnings. For example if the fault mask is [1,101,1] the warning would be printed 100 times. If we cache the invalid fault, it will only be printed once. This keeps the log more readable, the other solution at line 172 would flood the log with 100 warnings.

Reply:
Ok makes sense. Thanks!